File: src/get-type.ts
import { CommandType, SystemCommand } from "./cmd";

export default function getType(data: SystemCommand): CommandType {
    if (data.message.startsWith("!vc")) return CommandType.VimCommand;
    if (data.message.startsWith("!va")) return CommandType.VimAfter;
    if (data.message.startsWith("!vi")) return CommandType.VimInsert;
    if (data.commandType === CommandType.VimCommand) return CommandType.VimCommand;
    if (data.commandType === CommandType.VimInsert) return CommandType.VimInsert;
    if (data.commandType === CommandType.VimAfter) return CommandType.VimAfter;
    if (data.commandType === CommandType.SystemCommand) return CommandType.SystemCommand;
    if (data.commandType === CommandType.asdf) return CommandType.SystemCommand;
    if (data.commandType === CommandType.xrandr) return CommandType.SystemCommand;
    if (data.commandType === CommandType.i3Workspace) return CommandType.SystemCommand;
    if (data.commandType === CommandType.changeBackground) return CommandType.SystemCommand;

    return CommandType.VimAfter;
};

File: pkg/irc/getType.go
package irc

import "strings"

func GetType(msg IrcMessage) MessageType {
    if strings.HasPrefix(msg.Message, "!va") {
        return VimAfter
    } else if strings.HasPrefix(msg.Message, "!vi") {
        return VimInsert
    } else if strings.HasPrefix(msg.Message, "!vc") {
        return VimCommand
    }

	switch msg.Type {
	case VimAfter:
		return VimAfter
	case VimInsert:
		return VimInsert
	case VimCommand:
		return VimCommand
	case SystemCommand:
		return SystemCommand
	default:
		return SystemCommand
	}
}
